bcda/bcdacli/cli.go

```diff
@@ -598,19 +610,35 @@ func archiveExpiring(hrThreshold int) error {
 	for _, j := range jobs {
 		t := j.UpdatedAt
 		elapsed := time.Since(t).Hours()
-		if int(elapsed) >= hrThreshold {
-
-			id := int(j.ID)
-			jobDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_PAYLOAD_DIR"), id)
-			expDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_ARCHIVE_DIR"), id)
-
-			err = os.Rename(jobDir, expDir)
+		id := int(j.ID)
+		jobDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_PAYLOAD_DIR"), id)
+		_, err = os.Stat(jobDir)
+		jobDirExist := err == nil
+		expDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_ARCHIVE_DIR"), id)
+		_, err = os.Stat(expDir)
+		expDirExist := err == nil
```

Comment:
I think we can simplify this logic a little bit. If a job exists in the `FHIR_PAYLOAD_DIR`, we can make an assumption that it does not exist in the `FHIR_ARCHIVE_DIR`, and vise-versa.

With that assumption, I think we reduce the logic below into a single conditional. If the payloadDir exists, then move it to the archiveDir, then update the job status. That way we handle the error scenario where the payloadDir was moved to the archiveDir, but the job failed to update in the database.

snippet (w/o error handling)
```
if payloadDirExist {
    os.Rename(...)
    ...
}
db.Model(&j).Update("status",  models.JobStatusArchive)
```

Reply:
I agree on simplifying the logic to a single, but do you think we should look at the archive dir instead though?  If the original condition of this issue was that archive dirs were being orphaned then causing the rename to fail, should we check for those first then move on instead of assuming that if the payload dir exists try to copy.  Wouldn't that put us in the same situation where the job is marked completed, the payload file is created then the rename fails because the orphaned archive dir exists?

Reply:
From offline conversation I was looking for the wrong failing condition.  Updated the code to follow this pattern and look for the FHIR_PAYLOAD_DIR existing.  Was able to reduce the conditional and variable creation.